[stable/stolon] Fix issue with secret deletion during helm upgrade

rtluckie opened the pull request:
- fix issue with secret deletion during helm upgrade
- cleanup white-space in secrets
- add link to inspiration chart
- add lwolf as chart maintainer

Commit:
Fix issue with secret deletion during helm upgrade

- fix issue with secret deletion during helm update
- add link to inspiration chart
- add lwolf as chart maintainer

rtluckie commented:
/assign @unguiculus

Commit:
Add port definition to sentinel deployment manifest

- fix for prometheus scraping

unguiculus commented on stable/stolon/Chart.yaml:
```diff
@@ -1,12 +1,15 @@
 name: stolon
-version: 0.1.0
+version: 0.1.1
```
You should bump the minor version since you're adding features.

unguiculus commented on stable/stolon/templates/secret.yaml:
```diff
@@ -8,18 +7,17 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/resource-policy": keep
```
Can you explain why you remove `{{ if .Release.IsInstall }}` and add the annotation instead?

rtluckie replied:
@unguiculus  I'm still trying to sort this out.  Intent is to create the secret on initial install and maintain the secret data with subsequent `helm upgrade`s.   With `{{ if .Release.IsInstall }}` the secret is deleted during `helm upgrade`.  I thought the removal of `{{ if .Release.IsInstall }}` + addition of annotation would remedy, but it does not appear so.

see [gist](https://gist.github.com/rtluckie/7ee6dac40ce6cca91c78fbbb4196bde1) for details.  

any guidance would be appreciated.

unguiculus replied:
The annotation doesn't help here because you don't delete the release.

I would remove `{{ if .Release.IsInstall }}` and live with the fact that auto-generated passwords are regenerated on upgrade. Many charts have this problem. Using `--reuse-values` could help here. Alternatively, remove default passwords completely and make them `required`. You can provide a custom values file with passwords for CI only.

See https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values.

unguiculus commented on stable/stolon/templates/sentinel-deployment.yaml:
```diff
@@ -56,6 +56,11 @@ spec:
             - name: STSENTINEL_DEBUG
               value: {{ .Values.debug | quote}}
+          ports:
+{{- range $key, $value := .Values.ports }}
```
Needs to go into the readme.

Commit:
requested changes

Commit:
Suggested changes

- remove `"helm.sh/resource-policy": keep` annotation
- require `superuserPassword` and `replicationPassword`

rtluckie commented:
/retest

unguiculus commented on stable/stolon/values.yaml:
```diff
@@ -34,13 +34,13 @@ serviceAccount:
-## password for the superuser (default is 40 random chars)
-# superuserPassword: ""
+## password for the superuser
+superuserPassword: "superuserPassword"
```
Now you are setting default passwords. That's not good. If you don't want to auto-generate them, make them empty and use the `required` function. Provide an additional values.yaml for CI that sets passwords for CI only.

Commit:
Requested changes

- make `superuserPassword` and `replicationPassword` values required
- add CI specific values files
- bump to latest tagged image of stolon

unguiculus commented:
The values file for CI should only contain values that override the defaults from `values.yaml`. Please fix linting issues.

rtluckie commented:
@unguiculus all issues/requested changes _should_ be resolved

unguiculus committed:
Update values.yaml

unguiculus committed:
Update Chart.yaml

unguiculus committed:
Update README.md

unguiculus commented:
/lgtm
